freshmaker/pyxis.py

```diff
@@ -190,48 +190,39 @@ def get_latest_bundles(self, index_images):
 
         return ret_bundles
 
-    def get_digests_by_nvrs(self, nvrs):
+    def get_digest_by_nvr(self, nvr):
         """
-        Get images' digests(manifest_list_digest field) by their NVRs
+        Get image's digest(manifest_list_digest field) by its NVR
 
-        :param set nvrs: set of NVRs of ContainerImages to query Pyxis
-        :return: digests of images which we get by NVRs
-        :rtype: set
+        :param str nvr: NVR of ContainerImage to query Pyxis
+        :return: digest of image or None if manifest_list_digest not exists
+        :rtype: str or None
         """
         request_params = {'include': ','.join(['data.brew', 'data.repositories'])}
 
-        # get all manifest_list_digest of ContainerImages we got from Pyxis
-        digests = set()
-        for nvr in nvrs:
-            for image in self._pagination(f'images/nvr/{nvr}', request_params):
-                for repo in image.get('repositories'):
-                    if repo['published'] and 'manifest_list_digest' in repo:
-                        digests.add(repo['manifest_list_digest'])
-                        break
-        return digests
-
-    def filter_bundles_by_related_image_digests(self, original_digests,
-                                                bundles):
+        # get manifest_list_digest of ContainerImage from Pyxis
+        for image in self._pagination(f'images/nvr/{nvr}', request_params):
+            for repo in image.get('repositories'):
+                if repo['published'] and 'manifest_list_digest' in repo:
+                    return repo['manifest_list_digest']
+        return None
+
+    def get_bundles_by_related_image_digest(self, image_digest, bundles):
         """
-        Filter out bundles that don't have at least one 'related_image'
-        with the same manifest list digest as those in 'original_digests'
+        Get latest bundles that have the specified image digest in related images.
 
-        :param set original_digests: digests of the original
-            operator/operand (related image) images to filter bundles by them
-        :param list bundles: bundles to filter
-        :return: filtered list of bundles
+        :param str image_digest: digest of related image
+        :param list bundles: list of bundles to search from
+        :return: list of bundles
         :rtype: list
         """
-        ret_bundles = []
-        # If bundle doesn't have any of digests of images, don't add it to return list
+        ret = []
         for bundle in bundles:
-            for image in bundle.get('related_images', []):
-                # If same digest within images' digests is found, it will be in return list
-                if image.get('digest') in original_digests:
-                    ret_bundles.append(bundle)
-                    break
+            related_digests = [img.get('digest') for img in bundle.get('related_images', [])]
+            if image_digest in related_digests:
```

Comment:
This would avoid looping through all the related images when there is a match.
```suggestion
            if any (image_digest == img.get('digest') for img in bundle.get('related_images', [])):
```